extensions/azure/assetindex-cosmos/src/test/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetIndexIntegrationTest.java

```diff
@@ -0,0 +1,157 @@
+        CosmosContainerResponse containerIfNotExists = database.createContainerIfNotExists(CONTAINER_NAME, "/partitionKey");
+        container = database.getContainer(containerIfNotExists.getProperties().getId());
+        typeManager = new TypeManager();
+        typeManager.registerTypes(Asset.class, AssetDocument.class);
+        String partitionKey = "testpartition";
+        assetIndex = new CosmosAssetIndex(container, partitionKey, typeManager, new RetryPolicy<>());
+    }
+
+    @Test
+    void queryAssets_selectAll() {
+        Asset asset1 = Asset.Builder.newInstance()
+                .id("123")
+                .property("hello", "world")
+                .build();
+
+        Asset asset2 = Asset.Builder.newInstance()
+                .id("456")
+                .property("foo", "bar")
+                .build();
+
+
+        container.createItem(asset1);
+        container.createItem(asset2);
+
+        List<Asset> assets = assetIndex.queryAssets(AssetSelectorExpression.SELECT_ALL).collect(Collectors.toList());
+
+        assertThat(assets).hasSize(2)
+                .anyMatch(asset -> asset.getProperties().equals(asset1.getProperties()))
+                .anyMatch(asset -> asset.getProperties().equals(asset2.getProperties()));
+    }
+
+    @Test
+    void queryAssets_filterOnProperty() {
+        Asset asset1 = Asset.Builder.newInstance()
+                .id("123")
+                .property("test", "world")
+                .build();
+
+        Asset asset2 = Asset.Builder.newInstance()
+                .id("456")
+                .property("test", "bar")
+                .build();
+
+
+        container.createItem(asset1);
+        container.createItem(asset2);
+
+        AssetSelectorExpression expression = AssetSelectorExpression.Builder.newInstance()
+                .whenEquals("test", "bar")
+                .build();
+
+        List<Asset> assets = assetIndex.queryAssets(expression).collect(Collectors.toList());
+
+        assertThat(assets).hasSize(1)
+                .allSatisfy(asset -> {
+                    assertThat(asset.getId()).isEqualTo("456");
+                });
+    }
+
+    @Test
+    void findById() {
+        Asset asset1 = Asset.Builder.newInstance()
+                .id("123")
+                .property("test", "world")
+                .build();
+
+        Asset asset2 = Asset.Builder.newInstance()
+                .id("456")
+                .property("test", "bar")
+                .build();
+
+        container.createItem(asset1);
+        container.createItem(asset2);
+
+        Asset asset = assetIndex.findById("456");
+
+        assertThat(asset).isNotNull();
+        assertThat(asset.getProperties()).isEqualTo(asset2.getProperties());
+    }
+
+    @AfterEach
+    void teardown() {
+        CosmosContainerResponse delete = container.delete();
+        assertThat(delete.getStatusCode()).isGreaterThanOrEqualTo(200).isLessThan(300);
+    }
```

Comment:
I'd avoid deleting infrastructure within the integration test.

Reply:
If we don't then it means some temporary container will stay there after the integration test execution. 
I would rather clean up everything once test is properly executed, as it is done there.
@paullatzelsperger  @MoritzKeppler  what is your opinion?

Reply:
@paullatzelsperger I am not sure about your final answer :)
Do I keep it as it is?